models/classes/runner/time/AdjustmentMap.php

```diff
@@ -104,13 +108,13 @@ private function isValidAction(string $action): bool
         return in_array($action, [self::ACTION_INCREASE, self::ACTION_DECREASE], true);
     }
 
-    private function ensureEntryInitialized(string $sourceId)
+    private function ensureEntryInitialized(string $sourceId, string $type)
     {
-        if (!isset($this->map[$sourceId][self::ACTION_INCREASE])) {
-            $this->map[$sourceId][self::ACTION_INCREASE] = 0;
+        if (!isset($this->map[$sourceId][$type][self::ACTION_INCREASE])) {
```

Comment:
Could you divide this method to 2:
we can extract condition from here

Reply:
Done